.eslintrc.js

```diff
@@ -21,11 +28,278 @@ module.exports = {
 				filter: { regex: '^Model$', match: true },
 				format: ['PascalCase'],
 			},
-			...baseNamingRules,
+			// camelCase for everything not otherwise indicated
+			{ selector: 'default', format: ['camelCase'] },
+			// allow known default exclusions
+			{ selector: 'default', filter: { regex: '^(_id|__v|_raw)$', match: true }, format: null },
+			// allow variables to be camelCase or UPPER_CASE
+			{ selector: 'variable', format: ['camelCase', 'UPPER_CASE'] },
+			// allow known variable exclusions
+			{
+				selector: 'variable',
+				filter: { regex: '^(_id|__v|_raw)$', match: true },
+				format: null,
+			},
+			// GraphQL variables PascalCase
+			{
+				selector: 'variable',
+				filter: { regex: '^.*Gql$', match: true },
+				format: ['PascalCase'],
+			},
+			// do not enforce format on property names
+			{ selector: 'property', format: null },
+			// PascalCase for classes and TypeScript keywords
+			{
+				selector: ['typeLike'],
+				format: ['PascalCase'],
+			},
+			// allow trailing ASC and DESC on enumerations
+			{
+				selector: 'enumMember',
+				filter: { regex: '^.*?_(ASC|DESC)$', match: true },
+				format: null,
+			},
 		],
 	},
 
 	overrides: [
+		{
+			files: tsGlobs,
+			rules: {
+				// enforce return types on module boundaries
+				'@typescript-eslint/explicit-module-boundary-types': 'error',
+			},
+		},
+
+		{
+			files: ['**/*.types.ts', '**/types/*.ts', '**/*.schema.ts', '**/instances/**'],
```

Comment:
I think the `**/types/*.ts` file override is relevant for this repo, but I don't believe any of the others are.

Reply:
fixed.